articles/getting-started-with-inheritance-using-c/index.md

```diff
@@ -0,0 +1,228 @@
+Before we dig into the code it's important to note that we include the word **Override** when we need to reuse a method in a derived class inherited from a base class.
+
+If we are going to override a method in a derived class we can use the word **Virtual** before the method in a base class.
+
+Below we have an example where we calculate the area and perimeter of a circle using inheritance.
+
+In this piece of code, we'll define two private point coordinates, then we'll define a constructor that takes two integers from the main.
+
+**Note** that the Output statements use a reference (This) to implicitly call the ToString method.
+
+```c#
+    public class Point
+    {
+      private int x, y;
+
+      public Point( int xcoordinate, int ycoordinate )
+      {
+         X = xcoordinate;
+         Y = ycoordinate;
+         Console.WriteLine("point constructor: {0}", this);}
+  ```
+
+Here we define the set and get Accessors to access the private members of the class, then we'll override the ToString method.
+
+**Note** that the ToString method will return a string representation of the class Point.
+
+  ```c#
+      public int X
+      {
+         get
+         {return x;}
+         set
+         {x = value;}
+      }
+
+      public int Y
+      {
+         get
+         {return y;}
+         set
+         {y = value;}
+      }
+
+      public override string ToString()
+      {return "[" + X + ", " + Y + "]";}
+    } // end class Point
+   ```
+
+The following class is the Derived class of the class point.
+
+In the class Circle, we don't need to define the point coordinates because they've been inherited.
+
+`: base(xcoordinate,ycoordinate)`.
+
+Now we'll define the radius of the circle, the constructor, and the Accessors.
+
+   ```c#
+       public class Circle: Point
+    {
+       private double radius;
+
+      public Circle( int xcoordinate, int ycoordinate, double radiuscircle )
+         : base( xcoordinate, ycoordinate )
+      {Radius = radiuscircle;
+      Console.WriteLine("Circle constructor: {0}", this);}
+
+      public double Radius
+      {
+         get
+         {return radius;}
+         set
+         {radius = value;}
+      }
+
+   ```
+
+Now we need to define the functions that will help us calculate the area and perimeter of the circle, then we'll override the ToString method, just like we did in the class Point.
+
+   ```c#
+      public double Diameter()
+      {return Radius * 2;}
+
+      public double Perimeter()
+      {return Math.PI * Diameter();}
+
+
+      public virtual double Area()
+      {return Math.PI * Math.Pow( Radius, 2 );}
+
+      // Return string representation of Circle
+      public override string ToString()
+      {
+      // Using the ToString method with extra additions
+         return "Center= " + base.ToString() +"; Radius = " + Radius;   
+      }
+   } // end class Circle
+   ```
+
+Let's take a look at the following code:
+
+   ```c#
+   static void Main(string[] args)
```

Comment:
The above code has to be wrapped inside a class. I was getting an error with the code until I did this modification.

public class Example{

static void Main(string[] args)
    {
        Circle c = new Circle(6,8,4);
        Console.WriteLine("Area : " + c.Area());
        Console.WriteLine("Perimeter : " + c.Perimeter());
    }
}

Reply:
https://repl.it/@lalithNarayan/SoreMediumblueMemory#main.cs This is a repl link to the modification. Check it once and update your code accordingly.